### DISCUSSION TITLE
Power consumption question

### DISCUSSION THREAD

**Discussion by andersovreseth** [GMTrevis/Homeassistant-NodeRed-Systemair-VTR300#2] [Q&A]
Hello @GMTrevis ,

Great work with this capability for HA! I have set up up my own integration by reading your posts in various forums and it has been great help! I have managed to do most of the things I want, but so far I have not been able to read out the power consumption or any other numbers related to power. Which modbus registers are you using for this purpose? I can't seem to find any of them in the Systemair modbus documentation.

Hope you can give me some advise and pointers in the right direction.

BTW; I have a VTR 150 and my setup is using Modbus RTU with a 3-wire cable connected to a RS485-to-ethernet converter.

Best regards,
Anders

**Comment by GMTrevis** [comment]
Thanks, there is no power or energy data available from the ventilation unit itself.
I'm using TP-Link HS110 for monitoring the power consumption.
Delivered power is calculated/estimated in node red
The energy consumption in my config is accumulated daily, weekly, monthly and yearly using "Utility meter" integration in HA.
I hope this answers your questions :) 

![image](https://user-images.githubusercontent.com/69414078/150999014-35d31ede-ad8c-4967-9ed7-23b00e2cbb51.png)


`utility_meter:
  vtr300_kwh_cnt_daily:
    source: sensor.tp_link_vtr300_total_kwh
    cycle: daily       
  vtr300_kwh_cnt_weekly:
    source: sensor.tp_link_vtr300_total_kwh
    cycle: weekly    
  vtr300_kwh_cnt_monthly:
    source: sensor.tp_link_vtr300_total_kwh
    cycle: monthly  
  vtr300_kwh_cnt_yearly:
    source: sensor.tp_link_vtr300_total_kwh
    cycle: yearly   `

Code for estimated delivered power in node red:
`// hs := cp*ρ*q*dt   
// Eks.= (1.006 kJ/kg°C) (1.202 kg/m3) (1 m3/s) ((20°C) - (0°C)) 

//hs := sensible heat (kW)
//cp := specific heat of air (1.006 kJ/kg°C)
//p  := density of air (1.202 kg/m3)  or "press/(Rspecific*t)"
//q  := air volume flow (m3/s)
//dt := delta temperature delta [°C]
//press  := absolute pressure (Pa)
//t  := absolute temperature (kelvin)
//Rspecific := specific gas constant for dry air (J/(kg'K)) [Rspec]


var q_pv = flow.get ("vtr300_tilluftsvifte_pv","default");
var t_supply = flow.get ("vtr300_tillufts_temperatur","default");
var t_inlet = flow.get ("vtr300_inntak_temp","default");
var press = 101325;
var t = t_inlet + 273.15;
var Rspec = 287;

// Sjekker for "isNotNumber, ved "isNan" (opretting av Modbus kom./feil sensor data) tidlig "exit" (prÝ��v igjen).
  if (isNaN(q_pv) || isNaN(t_supply) || isNaN(t_inlet) || isNaN(t)) {
    return;
  }

var cp = 1.006;
//var p = 1.202; 
var p = press/(Rspec*t);
var q = (q_pv * 3)/3600;  
var dt = t_supply - t_inlet;
var hs = cp*p*q*dt;

// Lagrer watt og kW til memory for videreberegning av faktisk levert estimert effekt
flow.set("vtr300_tilfort_Watt", hs*1000,"default");  // Lagrer tilført kW til "memory/default"
flow.set("vtr300_tilfort_kWatt", hs,"default");  // Lagrer tilført kW til "memory/default"

// Estimert levert effekt & energi total
var watt = {payload:(hs*1000).toFixed(0)};
var kW = {payload:(hs).toFixed(2)};
var kW_10s = {payload:(hs/360)}; // kW/10s

// Tilpasser desimaler for attribute visning i HA
flow.set("vtr300_tilfort_Watt_formated", (hs*1000).toFixed(0) +' Watt',"default");  // Lagrer tilført kW til "memory/default"
flow.set("vtr300_tilfort_kWatt_formated", (hs).toFixed(2) +' kW',"default");  // Lagrer tilført kW til "memory/default"

return [watt,kW,kW_10s];`

Br
GM
